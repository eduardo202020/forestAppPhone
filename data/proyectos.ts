// Definimos la interfaz para la planta con la nueva estructura
export interface Planta {
  nombre_cientifico: string;
  cantidad: number; // La cantidad es opcional
}

// Definimos la interfaz para el proyecto con sus propiedades
export interface Projecto {
  id: string;
  photo: string;
  name: string;
  location: { latitude: number; longitude: number };
  startDate: string;
  progress: number;
  area: number;
  region: string;
  descripcion: string;
  img: string[];
  density: number;
  plantas: Planta[];
  estado: 'En progreso' | 'Pendiente' | 'Completado';
}

// Datos de ejemplo para los proyectos con la nueva estructura de plantas
export const proyectosData: Projecto[] = [
  {
    id: '1',
    photo:
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
    name: 'Reforestación Amazonas',
    startDate: '2023-05-01',
    region: 'Amazonas',
    location: { latitude: -12.0464, longitude: -77.0428 },
    progress: 45,
    area: 500,
    img: [
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
      'https://imgs.mongabay.com/wp-content/uploads/sites/20/2023/06/21125149/2_Miyawaki-in-Central-Delhi-1200x800.jpg',
    ],
    density: 3.8,
    descripcion:
      'Proyecto de reforestación para restaurar la biodiversidad en la región amazónica.',
    plantas: [
      {
        nombre_cientifico: 'Cedrelinga cateniformis',
        cantidad: 42,
      },
      {
        nombre_cientifico: 'Swietenia macrophylla',
        cantidad: 26,
      },
    ],
    estado: 'En progreso',
  },
  {
    id: '2',
    photo:
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
    name: 'Reforestación Loreto',
    startDate: '2023-06-15',
    region: 'Loreto',
    location: { latitude: -12.0464, longitude: -77.0428 },
    progress: 45,
    area: 500,
    img: [
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
      'https://imgs.mongabay.com/wp-content/uploads/sites/20/2023/06/21125149/2_Miyawaki-in-Central-Delhi-1200x800.jpg',
    ],
    density: 3.8,
    descripcion:
      'Reforestación en la zona norte de Loreto, con enfoque en la recuperación del ecosistema local.',
    plantas: [
      {
        nombre_cientifico: 'Dipteryx micrantha',
        cantidad: 31,
      },
      {
        nombre_cientifico: 'Bertholletia excelsa',
        cantidad: 22,
      },
    ],
    estado: 'Pendiente',
  },
  {
    id: '3',
    photo:
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
    name: 'Proyecto Puno',
    startDate: '2023-07-20',
    region: 'Puno',
    location: { latitude: -12.0464, longitude: -77.0428 },
    progress: 45,
    area: 500,
    img: [
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
      'https://imgs.mongabay.com/wp-content/uploads/sites/20/2023/06/21125149/2_Miyawaki-in-Central-Delhi-1200x800.jpg',
    ],
    density: 3.8,
    descripcion:
      'Reforestación en las laderas de Puno, con especies adaptadas a climas fríos.',
    plantas: [
      {
        nombre_cientifico: 'Queñua',
        cantidad: 10,
      },
      {
        nombre_cientifico: 'Buddleja coriacea',
        cantidad: 15,
      },
    ],
    estado: 'Completado',
  },
  {
    id: '4',
    photo:
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
    name: 'Restauración Madre de Dios',
    startDate: '2023-08-05',
    region: 'Madre de Dios',
    location: { latitude: -12.0464, longitude: -77.0428 },
    progress: 45,
    area: 500,
    img: [
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
      'https://imgs.mongabay.com/wp-content/uploads/sites/20/2023/06/21125149/2_Miyawaki-in-Central-Delhi-1200x800.jpg',
    ],
    density: 3.8,
    descripcion:
      'Proyecto para restaurar las zonas de bosque degradadas en la región de Madre de Dios.',
    plantas: [
      {
        nombre_cientifico: 'Dipteryx odorata',
        cantidad: 20,
      },
      {
        nombre_cientifico: 'Bertholletia excelsa',
        cantidad: 30,
      },
    ],
    estado: 'En progreso',
  },
];
