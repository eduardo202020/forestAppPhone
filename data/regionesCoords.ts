// Coordenadas aproximadas de las regiones para centrar el mapa
export const regionesCoords: Record<
  string,
  { latitude: number; longitude: number }
> = {
  Loreto: { latitude: -3.7491, longitude: -73.2538 },
  Cusco: { latitude: -13.5319, longitude: -71.9675 },
  Puno: { latitude: -15.8402, longitude: -70.0219 },
  'Madre de Dios': { latitude: -12.5933, longitude: -69.1871 },
  Amazonas: { latitude: -6.231, longitude: -77.8707 },
  'San Mart√≠n': { latitude: -7.0, longitude: -76.0 },
};
