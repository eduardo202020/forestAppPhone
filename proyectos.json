[
    {
        "id": "1",
        "nombre": "Reforestación Amazonas",
        "ubicacion": "Zona A, Amazonas",
        "fecha": "2023-05-01",
        "region": "Amazonas",
        "descripcion": "Proyecto de reforestación para restaurar la biodiversidad en la región amazónica.",
        "plantas": [
            {
                "nombre": "Cecropia",
                "tipo": "Árbol",
                "cantidad": 100
            },
            {
                "nombre": "Guaraná",
                "tipo": "Arbusto",
                "cantidad": 50
            }
        ],
        "estado": "En progreso"
    },
    {
        "id": "2",
        "nombre": "Reforestación Loreto",
        "ubicacion": "Zona B, Loreto",
        "fecha": "2023-06-15",
        "region": "Loreto",
        "descripcion": "Reforestación en la zona norte de Loreto, con enfoque en la recuperación del ecosistema local.",
        "plantas": [
            {
                "nombre": "Kapok",
                "tipo": "Árbol",
                "cantidad": 200
            },
            {
                "nombre": "Achiote",
                "tipo": "Arbusto",
                "cantidad": 75
            }
        ],
        "estado": "Pendiente"
    },
    {
        "id": "3",
        "nombre": "Proyecto Puno",
        "ubicacion": "Zona C, Puno",
        "fecha": "2023-07-20",
        "region": "Puno",
        "descripcion": "Reforestación en las laderas de Puno, con especies adaptadas a climas fríos.",
        "plantas": [
            {
                "nombre": "Queñua",
                "tipo": "Árbol",
                "cantidad": 150
            },
            {
                "nombre": "Matico",
                "tipo": "Planta medicinal",
                "cantidad": 60
            }
        ],
        "estado": "Completado"
    },
    {
        "id": "4",
        "nombre": "Restauración Madre de Dios",
        "ubicacion": "Zona D, Madre de Dios",
        "fecha": "2023-08-05",
        "region": "Madre de Dios",
        "descripcion": "Proyecto para restaurar las zonas de bosque degradadas en la región de Madre de Dios.",
        "plantas": [
            {
                "nombre": "Cacao",
                "tipo": "Árbol",
                "cantidad": 250
            },
            {
                "nombre": "Heliconia",
                "tipo": "Planta ornamental",
                "cantidad": 80
            }
        ],
        "estado": "En progreso"
    },
    {
        "id": "5",
        "nombre": "Reforestación San Martín",
        "ubicacion": "Zona E, San Martín",
        "fecha": "2023-09-10",
        "region": "San Martín",
        "descripcion": "Recuperación de áreas afectadas por la deforestación en la región de San Martín.",
        "plantas": [
            {
                "nombre": "Shihuahuaco",
                "tipo": "Árbol",
                "cantidad": 180
            },
            {
                "nombre": "Palo de rosa",
                "tipo": "Árbol",
                "cantidad": 100
            }
        ],
        "estado": "Pendiente"
    }
]