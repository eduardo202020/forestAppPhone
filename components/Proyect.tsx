import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import React from 'react';
import { Card, ProgressBar } from 'react-native-paper';
// import FontAwesome from '@expo/vector-icons/FontAwesome';

import { Projecto } from '@/data/proyectos';

const Proyect: React.FC<{ proyect: Projecto }> = ({ proyect }) => {
  return (
    <Card key={proyect.id} style={styles.projectCard}>
      {/* Contenedor para el ícono flotante */}
      <View style={styles.favoriteIconContainer}>
        {/* <FontAwesome name="heart-o" size={24} color="#e74c3c" /> */}
      </View>
      <Card.Title
        title={proyect.name}
        titleStyle={{ fontSize: 20 }}
        subtitle={`Iniciado el ${proyect.startDate}`}
        subtitleStyle={{ fontSize: 14, fontWeight: 'bold' }}
      />
      <Card.Content style={{ flexDirection: 'column', gap: 10 }}>
        <ProgressBar progress={proyect.progress / 100} color="#2ecc71" />
        <Text style={styles.projectText}>Área: {proyect.area} m²</Text>
        <Text style={styles.projectText}>
          Densidad: {proyect.density} plantas/m²
        </Text>
        <Card.Cover
          source={{ uri: proyect.img[0] }}
          style={styles.plantImage}
        />
        <Card.Cover
          source={{ uri: proyect.img[1] }}
          style={styles.plantImage}
        />
      </Card.Content>

      <Card.Actions>
        <TouchableOpacity>
          <Text style={styles.detailsText}>Ver detalles</Text>
        </TouchableOpacity>
      </Card.Actions>
    </Card>
  );
};

export default Proyect;

const styles = StyleSheet.create({
  projectCard: {
    margin: 10,
    position: 'relative', // Asegura que el ícono flotante se posicione relativo al Card
  },
  favoriteIconContainer: {
    position: 'absolute',
    top: 10,
    right: 10,
    zIndex: 1, // Asegura que el ícono esté por encima del contenido
    borderRadius: 20,
    padding: 5,
  },
  projectText: {
    fontSize: 16,
  },
  plantImage: {
    height: 150,
    marginVertical: 10,
  },
  detailsText: {
    color: '#3498db',
    fontWeight: 'bold',
  },
});
