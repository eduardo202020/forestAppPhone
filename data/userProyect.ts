import { Projecto } from '@/data/proyectos';

export const userProyects: Projecto[] = [
  {
    id: '1',
    photo:
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',
    name: 'Bosque Urbano Loreto',
    location: { latitude: -12.0464, longitude: -77.0428 },
    startDate: '2025-01-10',
    progress: 45,
    area: 500,
    region: 'Loreto',
    descripcion:
      'Proyecto de reforestaci칩n para restaurar la biodiversidad en la regi칩n amaz칩nica.',
    img: [
      'https://www.ecolandscaping.org/wp-content/uploads/2021/12/PostPlanting_DinoKuznik-scaled.jpeg',

      'https://imgs.mongabay.com/wp-content/uploads/sites/20/2023/06/21125149/2_Miyawaki-in-Central-Delhi-1200x800.jpg',
    ],
    density: 3.8,
    plantas: [
      { nombre_cientifico: 'Tornillo', cantidad: 42 },
      { nombre_cientifico: 'Caoba', cantidad: 26 },
      { nombre_cientifico: 'Shihuahuaco', cantidad: 31 },
      { nombre_cientifico: 'Casta침o', cantidad: 22 },
      { nombre_cientifico: 'Cedar', cantidad: 45 },
    ],
    estado: 'En progreso',
  },
];
