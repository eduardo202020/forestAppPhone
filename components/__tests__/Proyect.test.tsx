import * as React from 'react';
import renderer from 'react-test-renderer';
import Proyect from '@/components/Proyect'; // Ajusta la ruta si es necesario
import { Projecto } from '@/data/proyectos';

// Datos de ejemplo para el test
const proyectoEjemplo: Projecto = {
  id: '1',
  name: 'Proyecto de Reforestación',
  startDate: '2023-01-01',
  progress: 50,
  area: 1000,
  density: 10,
  img: ['https://example.com/img1.jpg', 'https://example.com/img2.jpg'],
};

it('debería renderizar correctamente el componente Proyect', () => {
  const tree = renderer.create(<Proyect proyect={proyectoEjemplo} />).toJSON();
  expect(tree).toMatchSnapshot(); // Comparar con el snapshot
});
