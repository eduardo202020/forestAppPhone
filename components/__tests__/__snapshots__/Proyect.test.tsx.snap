// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debería renderizar correctamente el componente Proyect 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(247, 243, 249, 1)",
      "borderRadius": 12,
      "margin": 10,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        borderRadiusStyles={{}}
        index={0}
        siblings={
          [
            "View",
            "Card.Title",
            "Card.Content",
            "Card.Actions",
          ]
        }
        style={
          {
            "borderRadius": 20,
            "padding": 5,
            "position": "absolute",
            "right": 10,
            "top": 10,
            "zIndex": 1,
          }
        }
        total={4}
      />
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "minHeight": 30,
                      "paddingRight": 16,
                    },
                    {
                      "marginBottom": 0,
                    },
                    {
                      "fontSize": 20,
                    },
                  ],
                ],
              ]
            }
          >
            Proyecto de Reforestación
          </Text>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginVertical": 0,
                      "minHeight": 20,
                      "paddingRight": 16,
                    },
                    {
                      "fontSize": 14,
                      "fontWeight": "bold",
                    },
                  ],
                ],
              ]
            }
          >
            Iniciado el 2023-01-01
          </Text>
        </View>
        <View />
      </View>
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            undefined,
            {
              "flexDirection": "column",
              "gap": 10,
            },
          ]
        }
      >
        <View
          accessibilityRole="progressbar"
          accessibilityState={
            {
              "busy": true,
            }
          }
          accessibilityValue={
            {
              "max": 100,
              "min": 0,
              "now": 50,
            }
          }
          accessible={true}
          onLayout={[Function]}
          style={false}
          testID="progress-bar"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(231, 224, 236, 1)",
                "height": 4,
                "opacity": 0,
                "overflow": "hidden",
              }
            }
          />
        </View>
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Área: 
          1000
           m²
        </Text>
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Densidad: 
          10
           plantas/m²
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              {
                "borderRadius": 12,
              },
              {
                "height": 150,
                "marginVertical": 10,
              },
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={
              {
                "uri": "https://example.com/img1.jpg",
              }
            }
            style={
              [
                {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "width": undefined,
                },
                {
                  "borderRadius": 12,
                },
              ]
            }
          />
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              {
                "borderRadius": 12,
              },
              {
                "height": 150,
                "marginVertical": 10,
              },
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={
              {
                "uri": "https://example.com/img2.jpg",
              }
            }
            style={
              [
                {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "width": undefined,
                },
                {
                  "borderRadius": 12,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        borderRadiusStyles={{}}
        index={3}
        siblings={
          [
            "View",
            "Card.Title",
            "Card.Content",
            "Card.Actions",
          ]
        }
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 8,
            },
            undefined,
            {
              "justifyContent": "flex-end",
            },
          ]
        }
        total={4}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginLeft": 8,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#3498db",
                "fontWeight": "bold",
              }
            }
          >
            Ver detalles
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
